spec/support/shared/examples/metasploit/cache/module/descendant.rb

```diff
@@ -0,0 +1,17 @@
+shared_examples_for 'Metasploit::Cache::Module::Descendant' do |ancestor:, factory:|
+  context 'associations' do
+    it { is_expected.to belong_to(:ancestor).class_name(ancestor.fetch(:class_name)).inverse_of(ancestor.fetch(:inverse_of)) }
+  end
+
+  context 'validations' do
+    it { is_expected.to validate_presence_of(:ancestor) }
+
+    context 'with existing record' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.